packages/runtime-core/src/apiWatch.ts

```diff
@@ -190,6 +190,15 @@ function doWatch(
     }
   }
 
+  // TODO remove in 3.5
+  if (__DEV__ && deep !== void 0 && typeof deep !== 'boolean') {
```

Comment:
I think we should be more explicit: only check for number values, and mention in the warning that numbers will be used to indicate watch depth in a future version.

Reply:
done